src/swapd/process_msgs.rs

```diff
@@ -0,0 +1,1043 @@
+                );
+                false
+            }
+        };
+        // if updated_state {
+        //     self.save_state()?;
+        //     info!(
+        //         "ChannelStateMachine {} switched to {} state",
+        //         self.state().channel.active_channel_id(),
+        //         self.state().state_machine
+        //     );
+        // }
+        Ok(updated_state)
+    }
+
+    fn process_event(&mut self, event: Event<Request>) -> Result<(), Error> {
+        match Awaiting::with(
+            self.state(),
+            &event,
+            Some(self.syncer_state()),
+            Some(&self.pending_requests),
+        ) {
+            // Msg bus
+            Awaiting::TakerP2pMakerCommit => self.taker_p2p_maker_commit(event),
+            Awaiting::MakerTakerP2pRevealProof => self.maker_taker_p2p_reveal_proof(event),
+            Awaiting::MakerTakerP2pReveal => self.maker_taker_p2p_reveal(event),
+            Awaiting::AliceP2pCoreArb => self.alice_p2p_core_arb(event),
+            Awaiting::AliceP2pBuySig => self.alice_p2p_buy_sig(event),
+            Awaiting::BobP2pRevealAlice => self.bob_p2p_reveal_alice(event),
+            Awaiting::BobP2pRefundSigs => self.bob_p2p_refund_sigs(event),
+
+            // Ctl bus
+            Awaiting::TakerCtlTakeSwap => self.taker_ctl_take_swap(event),
+            Awaiting::MakerCtlRevealProof => self.maker_ctl_reveal_proof(event),
+            Awaiting::MakerCtlMakeSwap => self.maker_ctl_make_swap(event),
+            Awaiting::MakerTakerCtlFundingUpdated => self.maker_taker_ctl_funding_updated(event),
+            Awaiting::BobCtlCoreArb => self.bob_ctl_core_arb(event),
+            Awaiting::AliceCtlRefundSigs => self.alice_ctl_refund_sigs(event),
+            Awaiting::BobCtlBuySig => self.bob_ctl_buy_sig(event),
+            Awaiting::BobCtlTxLock => self.bob_ctl_tx_lock(event),
+            Awaiting::CtlTx => self.ctl_tx(event),
+            Awaiting::CtlSweepMonero => self.ctl_sweep_monero(event),
+            Awaiting::CtlTerminate => self.ctl_terminate(),
+            Awaiting::CtlSweepBitcoin => self.ctl_sweep_bitcoin(event),
+            Awaiting::CtlPeerReconnected => self.ctl_peer_reconnected(event),
+            Awaiting::CtlCheckpoint => self.ctl_checkpoint(event),
+
+            // Ctl + Rpc bus
+            Awaiting::CtlRpcAbortSimple => self.ctl_rpc_abort_simple(event),
+            Awaiting::BobCtlRpcAbortBob => self.bob_ctl_rpc_abort_bob(event),
+            Awaiting::CtlRpcAbortBlocked => self.ctl_rpc_abort_blocked(event),
+
+            // Rpc bus
+            Awaiting::RpcGetInfo => self.rpc_get_info(event),
+
+            // Syncer bus
+
+            // Others
+            Awaiting::Unknown => Err(Error::Farcaster(s!("Invalid state for this event"))),
+        }
+    }
+}
+
+/// Ctl bus processing fns
+impl Runtime {
+    fn taker_ctl_take_swap(&mut self, event: Event<Request>) -> Result<(), Error> {
+        if let Request::TakeSwap(InitSwap {
```

Comment:
I am not happy with this matching. The event matching should happen one layer further up. `taker_ctl_take_swap` can only process `Request::TakeSwap`, so make that the argument. This also applies to all the functions following this one. Their interfaces basically invalidate the work you already did at their call site.